dlt/destinations/athena/athena.py

```diff
@@ -0,0 +1,289 @@
+
+    def drop_tables(self, *tables: str) -> None:
+        if not tables:
+            return
+        statements = [f"DROP TABLE IF EXISTS {self.capabilities.escape_identifier(table)};" for table in tables]
+        self.execute_fragments(statements)
+
+    @contextmanager
+    @raise_database_error
+    def begin_transaction(self) -> Iterator[Any]:
+        yield self
+
+    @staticmethod
+    def _make_database_exception(ex: Exception) -> Exception:
+        print(str(ex))
+        if isinstance(ex, OperationalError):
+            if "TABLE_NOT_FOUND" in str(ex):
+                return DatabaseUndefinedRelation(ex)
+            elif "SCHEMA_NOT_FOUND" in str(ex):
+                return DatabaseUndefinedRelation(ex)
+            elif "Table not found" in str(ex):
+                return DatabaseUndefinedRelation(ex)
+            elif "Database does not exist" in str(ex):
+                return DatabaseUndefinedRelation(ex)
+            return DatabaseTerminalException(ex)
+        elif isinstance(ex, (ProgrammingError, IntegrityError)):
+            return DatabaseTerminalException(ex)
+        if isinstance(ex, DatabaseError):
+            return DatabaseTransientException(ex)
+        return ex
+
+    def execute_sql(self, sql: AnyStr, *args: Any, **kwargs: Any) -> Optional[Sequence[Sequence[Any]]]:
+        with self.execute_query(sql, *args, **kwargs) as curr:
+            if curr.description is None:
+                return None
+            else:
+                f = curr.fetchall()
+                return f
+
+    @staticmethod
+    def _convert_to_old_pyformat(new_style_string: str, args: Tuple[Any, ...]) -> Tuple[str, Dict[str, Any]]:
+        # create a list of keys
+        keys = ["arg"+str(i) for i, _ in enumerate(args)]
+        # create an old style string and replace placeholders
+        old_style_string, count = re.subn(r"%s", lambda _: "%(" + keys.pop(0) + ")s", new_style_string)
+        # create a dictionary mapping keys to args
+        mapping = dict(zip(["arg"+str(i) for i, _ in enumerate(args)], args))
+        # raise if there is a mismatch between args and string
+        if count != len(args):
+            raise DatabaseTransientException(OperationalError())
+        return old_style_string, mapping
+
+    @contextmanager
+    @raise_database_error
+    def execute_query(self, query: AnyStr, *args: Any, **kwargs: Any) -> Iterator[DBApiCursor]:
+        assert isinstance(query, str)
+
+        db_args = kwargs
+        # convert sql and params to PyFormat, as athena does not support anything else
+        if args:
+            query, db_args = self._convert_to_old_pyformat(query, args)
+            if kwargs:
+                db_args.update(kwargs)
+        cursor = self._conn.cursor(formatter=DLTAthenaFormatter())
+        for query_line in query.split(";"):
+            if query_line.strip():
+                try: 
+                    cursor.execute(query_line, db_args)
```

Comment:
should we wrap this cursor in one that provides `df()` method? like postgres?

Reply:
it basically forwards most of the standard cursor methods to inner cursor so all shoudl work

Reply:
seems to work!